### PR TITLE
Changes to DBus and engine APIs for idempotence

### PR DESCRIPTION
**Pull request by jbaublitz**
Supersedes #1614 
Relates to stratis-storage/project#51

Make internal changes to the engine to better support idempotent operations using the DBus API so that the return value of DBus methods provides information on what was changed and omits resources that were not changed.

### CODE REVIEW

**Comment by mulkieran** [comment]
It looks like Travis is having trouble, I restarted some tasks.

**Comment by jbaublitz** [comment]
I was just about to comment on that. It's associated with the previous PR still. It's pulling from the wrong refs. Trying to figure out how to kick off a new build. I think I have to change the commit hash somehow.

**Comment by mulkieran** [comment]
That same problem happened to me just once w/ pyudev.

**Comment by jbaublitz** [comment]
New CI job kicked off. We should be okay now.

File: tests/client-dbus/src/stratisd_client_dbus/_data.py [hunk 1]
```diff
@@ -53,38 +53,38 @@
...
 <arg name="filesystems" type="ao" direction="in"/>
-<arg name="results" type="as" direction="out"/>
+<arg name="results" type="(bas)" direction="out"/>
 <arg name="return_code" type="q" direction="out"/>
 <arg name="return_string" type="s" direction="out"/>
 </method>
 <method name="SetName">
 <arg name="name" type="s" direction="in"/>
-<arg name="action" type="b" direction="out"/>
+<arg name="result" type="(bbs)" direction="out"/>
```
Comment (mulkieran) [inline tests/client-dbus/src/stratisd_client_dbus/_data.py]: Whoops! Signature is wrong.

Comment (jbaublitz) [reply tests/client-dbus/src/stratisd_client_dbus/_data.py]: Oh! I'll fix this now.

**Comment by jbaublitz** [comment]
@mulkieran I think I got them all.

**Comment by mulkieran** [comment]
Ok, re-approved.

**Pull request merged**